test/cql-pytest/test_bloom_filter.py

```diff
@@ -28,7 +28,7 @@ def test_bloom_filter(scylla_only, cql, test_keyspace, N, M, fp_chance):
         fp = rest_api.get_column_family_metric(cql, 
           "bloom_filter_false_positives", table)
         ratio = fp / M
-        assert ratio >= fp_chance * 0.7 and ratio <= fp_chance * 1.15
+        assert fp > 0 and ratio <= fp_chance * 1.15
```

Comment:
Can we instead make sure all test data is in a single group? If that is too difficult, it might not be worth it.

Reply:
I think this relaxes the test too much.

I don't understand why multiple compaction groups make the bloom filter more efficient. I guess they do, if there is a too-low number of partitions.

Reply:
```
pkeys		ssts		pks:sst		data		filter		data:filter
100		1		100		4300		136		31
100		10		10		4300		240		17
100		100		1		4300		1600		2
1000		1		1000		43872		1264		34
1000		10		100		43000		1360		31
1000		100		10		43000		2400		17
10000		1		10000		439872		12512		35
10000		10		1000		438720		12640		34
10000		100		100		430000		13600		31
100000		1		100000		4483488		125016		35
100000		10		10000		4398720		125120		35
100000		100		1000		4387200		126400		34
```

data above shows that with < 1k pkeys, data:filter ratio increases significantly as number of sstables increases.  with 10k keys, there's barely any difference. with 100k keys, even less.

we have two options, resample bloom filters to meet a data:filter ratio goal, but that's a long term effort.

alternatively, I can just increase number of keys to 10k, with the test time going from 1.17s to 4.39s. sounds good to you?

Reply:
An idea: run the test with 500 pkeys as today. On failure, switch to slow mode which will try again with 10k-100k pkeys. Of course, we're embedding implementation detail into test, which isn't the best thing to do, but that's a reasonable assumption that bloom filter should increase in efficiency if the underlying database stores the keys inserted into multiple bloom filters, one per sstable. So if first line of defense fails, we go into slow mode, but for regular runs, fast mode happens.